terraform/20-app/ecr.tf

```diff
@@ -41,6 +41,24 @@ module "ecr_front_end" {
   repository_lifecycle_policy = local.standard_ecr_lifecycle_policy
 }
 
+resource "terraform_data" "dummy_front_end_image_provisioner" {
+  depends_on = [module.ecr_front_end]
+  provisioner "local-exec" {
+    command = <<EOF
+      aws ecr get-login-password --region eu-west-2 | docker login --username AWS --password-stdin ${local.account_id}.dkr.ecr.eu-west-2.amazonaws.com
+      docker pull alpine
+      docker tag alpine ${module.ecr_front_end.repository_url}:null
+      docker push ${module.ecr_front_end.repository_url}:null
+    EOF
+  }
+}
+
+data "aws_ecr_image" "front_end" {
```

Comment:
This is all now very verbose and repetitive, but the general flow is:
- ECR to hold images
- Dummy image provisioner to populate the ECR with an initial alpine image
- data source to reference the most recent image not just the one with the `latest` tag

The provisioner is needed because when terraform plans the data source, the underlying query errors because there are no image artifacts in the ECR.

My plan is to build a re-useable terraform module after this and wrap all of this duplicated logic in there, because the pattern is the same across all of our ECRs